R/knitr-engine.R

```diff
@@ -53,7 +53,31 @@ eng_python <- function(options) {
   ranges <- mapply(c, starts, ends, SIMPLIFY = FALSE)
   
   # line index from which source should be emitted
-  pending <- 1
+  pending_source_index <- 1
+  
+  # plots captured from Python which need to be emitted
+  pending_plots <- list()
+  
+  # monkey-patch matplotlib's show method: rather than showing a plot
+  # on the screen, we write the plot to file
+  matplotlib <- import("matplotlib", convert = FALSE)
```

Comment:
Can we affect plot sizes somehow? (we get the knitr figure sizes passed to us in options)

Reply:
I think we should be able to; let me investigate a bit more.

Reply:
So this functionality depends on matplotlib?

Reply:
We probably want to do a check for matplotlib before executing this. You can use `py_module_available("matplotlib")` for this.

Reply:
`matplotlib` seems like the primary way through which Python users generate plots. Is there anything else we should consider supporting?

Reply:
Others are a bit too fancy and may introduce heavier dependencies. I think matplotlib should suffice for now.